src/lib/utils/functions.ts

```diff
@@ -96,13 +98,20 @@ export async function evaluateNft(tokenUri) {
       return new NftEvaluation(evaluation.image_grade, evaluation.image_grade, UriType.OnChain, evaluation.image_location, metadata);
     } catch (e) {
       console.error(e);
-      return new NftEvaluation(Grade.Unknown, Grade.Unknown, UriType.OnChain, ImageLocation.Unknown, null);
+      return new NftEvaluation(Grade.Unknown, Grade.Unknown, UriType.OnChain, ImageLocation.MetadataError, null);
     }
   } else {
     const [url, uriType, urlObj] = getResolvableUrl(tokenUri)
-    const metadataStr = await getMetadataFromUrl(url)
     const uriGrade = gradeUriType(uriType);
 
+    let metadataStr
+    try {
+      metadataStr = await getMetadataFromUrl(url)
+    } catch (e) {
+      console.log(e)
```

Comment:
prefer `console.error`

Reply:
whoops yeah i saw that in your changes, will fix in a sec